tempesta_fw/cache.c

```diff
@@ -634,6 +634,8 @@ tfw_cache_build_resp_hdr(TDB *db, TfwHttpResp *resp, TfwStr *hdr,
 	TfwStr *dups;
 	TfwCStr *s = (TfwCStr *)*p;
 
+	TFW_STR_INIT(hdr);
```

Comment:
In my opinion, If `hdr` is intended to be in initial, zeroed state, then that should be be done in the calling function, and not here. The initial state of `hdr` should be enforced then via a proper `BUG_ON()`.